src/client.rs

```diff
@@ -855,25 +854,45 @@ impl Client {
     /// Pull a single layer from an OCI registry.
     ///
     /// This pulls the layer for a particular image that is identified by
-    /// the given digest. The image reference is used to find the
+    /// the given layer descriptor. The image reference is used to find the
     /// repository and the registry, but it is not used to verify that
     /// the digest is a layer inside of the image. (The manifest is
     /// used for that.)
     pub async fn pull_blob<T: AsyncWrite + Unpin>(
         &self,
         image: &Reference,
-        digest: &str,
+        layer: &OciDescriptor,
         mut out: T,
     ) -> Result<()> {
-        let url = self.to_v2_blob_url(image.resolve_registry(), image.repository(), digest);
-        let mut stream = RequestBuilderWrapper::from_client(self, |client| client.get(&url))
-            .apply_accept(MIME_TYPES_DISTRIBUTION_MANIFEST)?
-            .apply_auth(image, RegistryOperation::Pull)?
-            .into_request_builder()
-            .send()
-            .await?
-            .error_for_status()?
-            .bytes_stream();
+        let mut url =
+            &self.to_v2_blob_url(image.resolve_registry(), image.repository(), &layer.digest);
+
+        let mut alternatives = if let Some(alternatives) = &layer.urls {
```

Comment:
I think to be spec compliant, we should be skipping any urls that don't have a `http` or `https` scheme. See https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#properties

Reply:
Fixed